Sources/Internal/WebSocketConnection.swift

```diff
@@ -35,25 +49,66 @@ class WebSocketConnection {
         self.onDisconnect = onDisconnect
         self.onTextReceive = onTextReceive
         serialCallbackQueue = DispatchQueue(label: "org.walletconnect.swift.connection-\(url.bridgeURL)-\(url.topic)")
-        socket = WebSocket(url: url.bridgeURL)
-        socket.delegate = self
-        socket.callbackQueue = serialCallbackQueue
+        
+        self.socket = WebSocket(request: URLRequest(url: url.bridgeURL), engine: WSEngine(transport: FoundationTransport(), certPinner: FoundationSecurity()))
+        self.socket.callbackQueue = serialCallbackQueue
+        self.socket.delegate = self
+        
+#if os(iOS)
+        var didBecomeActiveName: NSNotification.Name
+        if #available(iOS 13.0, *) {
+            didBecomeActiveName = UIScene.didActivateNotification
+        } else {
+            didBecomeActiveName = UIApplication.didBecomeActiveNotification
+        }
+        self.foregroundNotificationObserver = NotificationCenter.default.addObserver(forName: didBecomeActiveName,
+                                                                                     object: nil,
+                                                                                     queue: OperationQueue.main) { [weak self] notification in
+            self?.open()
+        }
+        var willResignActiveName: NSNotification.Name
+        if #available(iOS 13.0, *) {
+            willResignActiveName = UIScene.willDeactivateNotification
+        } else {
+            willResignActiveName = UIApplication.willResignActiveNotification
+        }
+        self.backgroundNotificationObserver = NotificationCenter.default.addObserver(forName: willResignActiveName,
+                                                                                     object: nil,
+                                                                                     queue: OperationQueue.main) { [weak self] notification in
+            print("WebSocketConnect: ==> Shotting down socket before background")
+            self?.close(closeCode: CloseCode.protocolError.rawValue)
+        }
+#endif
     }
-
+    
+    deinit {
+        LogService.shared.log("WebSocketConnect: ==> deinit")
+#if os(iOS)
+        if let observer = self.backgroundNotificationObserver {
+            NotificationCenter.default.removeObserver(observer)
+        }
+        if let observer = self.foregroundNotificationObserver {
+            NotificationCenter.default.removeObserver(observer)
+        }
+#endif
```

Comment:
```suggestion
```

I think there's no need to call removeObserver in `deinit` (https://developer.apple.com/documentation/foundation/notificationcenter/1413994-removeobserver)

Reply:
Thanks for checking!

I actually see the opposite example in:
https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver

And it is logical in a sense that somebody who registers to notifications holds the reference and has control over it.
Am I missing something?